src/tls/key_pair.h

```diff
@@ -670,15 +670,12 @@ namespace tls
       // Because mbedtls does not support parsing x509v3 extensions from a
       // CSR (https://github.com/ARMmbed/mbedtls/issues/2912), the CA sets the
       // SAN directly instead of reading it from the CSR
-      for (auto& subject_alt_name : subject_alt_names)
+      auto rc =
+        x509write_crt_set_subject_alt_names(&sign.crt, subject_alt_names);
```

Comment:
put a try/catch around this? We should not crash if we are given a malformed cert